integration/integration_test.go

```diff
@@ -313,3 +313,66 @@ func TestDemoteLeader(t *testing.T) {
 	numManager--
 	pollClusterReady(t, cl, numWorker, numManager)
 }
+
+func TestJoinDemoteRace(t *testing.T) {
```

Comment:
Oh, seems like it waits for the dispatcher heartbeat to timeout? I don't think we need that for this test. Could we just do a forced removal?